### PR TITLE
added lens subgraph

### PR DESCRIPTION
**Pull request by sharathkrml**

### CODE REVIEW

File: lens/src/mappings/index.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    let profileId = getDefaultProfileIdfromUser(event.params.from)
+    if (profileId == event.params.tokenId) {
+      log.debug('default profile reset ,hash {}', [txHash.toHexString()])
+      airstack.social.trackSocialUserDefaultProfileChange(
+        event.block,
+        txHash.toHexString(),
+        event.logIndex,
+        event.transaction.from.toHexString(),
+        event.address.toHexString(),
+        '',
```
Comment (rahul7668gupta) [inline lens/src/mappings/index.ts:118]: not sure if we should be passing empty strings for a tokenId, should be a seperate bool flag which can be used to check the condition.

File: lens/src/mappings/index.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  let txHash = event.transaction.hash
+  let defaultProfileSet = DefaultProfileSetEntity.load(
+    txHash
+      .toHexString()
+      .concat('-')
+      .concat(event.logIndex.toString()),
+  )
+  if (defaultProfileSet != null) {
+    log.error('defaultProfileSet already exists,hash {}', [txHash.toHexString()])
+    throw new Error('')
```
Comment (rahul7668gupta) [inline lens/src/mappings/index.ts:147]: should send a proper error message

File: lens/src/mappings/index.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    let lensNameArr = userToProfileMap.lensNames
+    if (lensNameArr == null) {
+      throw new Error('')
+    } else {
+      lensNameArr.push(event.params.handle)
+    }
+    userToProfileMap.lensNames = lensNameArr
+    let hashArr = userToProfileMap.hashes
+    if (hashArr == null) {
+      throw new Error('')
```
Comment (rahul7668gupta) [inline lens/src/mappings/index.ts:47]: should send a proper error message

File: lens/src/mappings/index.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    userToProfileMap.user = to
+    userToProfileMap.profile = [profileId]
+    userToProfileMap.lensNames = [event.params.handle]
+    userToProfileMap.count = BIGINT_ONE
+    userToProfileMap.hashes = [txHash]
+  } else {
+    log.error('one User multiple profile,txHash {}', [txHash])
+    let profileArr = userToProfileMap.profile
+    if (profileArr == null) {
+      throw new Error('')
```
Comment (rahul7668gupta) [inline lens/src/mappings/index.ts:33]: should send a proper error message

File: lens/src/mappings/index.ts [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+export function handleDefaultProfileSet(event: DefaultProfileSet): void {
+  let txHash = event.transaction.hash
+  let defaultProfileSet = DefaultProfileSetEntity.load(
+    txHash
+      .toHexString()
+      .concat('-')
+      .concat(event.logIndex.toString()),
+  )
+  if (defaultProfileSet != null) {
+    log.error('defaultProfileSet already exists,hash {}', [txHash.toHexString()])
```
Comment (rahul7668gupta) [inline lens/src/mappings/index.ts:146]: why are we throwing error here? can someone not set one profile as defualt and then change it to another profile of theirs?

**Review by rahul7668gupta — CHANGES_REQUESTED** [review]

File: lens/tests/lens.test.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  test('Testing minting,setting default profile,transferring & automatic unsetting', () => {
+    let profileCreatedEvent = getProfileCreatedEvent(profileCreated1)
+
+    handleProfileCreated(profileCreatedEvent)
+    let defaultProfileSetEvent = getDefaultProfileSet(setDefault2)
+    handleDefaultProfileSet(defaultProfileSetEvent)
+
+    let transferEvent = getTransferEvent(transfer3)
+
+    handleTransfer(transferEvent)
```
[the remark this answers is not in the document]
Comment (sharathkrml) [reply lens/tests/lens.test.ts]: Done

File: lens/src/mappings/index.ts [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    event.logIndex,
+    event.transaction.from.toHexString(),
+    event.address.toHexString(),
+    event.params.profileId.toString(),
+    LENSHUB_ADDRESS.toHexString(),
+    event.params.wallet.toHexString(),
+  )
+}
+
+function createOrUpdateUserToDefaultProfileIdMap(wallet: Bytes, profileId: BigInt): void {
```
[the remark this answers is not in the document]
Comment (sharathkrml) [reply lens/src/mappings/index.ts:170]: done

File: lens/src/mappings/index.ts (hunk 5 above)
Comment (sharathkrml) [reply lens/src/mappings/index.ts:143]: just for logging - removed

File: lens/src/mappings/index.ts [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+  UserToDefaultProfileIdMap,
+} from '../../generated/schema'
+
+export function handleProfileCreated(event: ProfileCreated): void {
+  let txHash = event.transaction.hash.toHexString()
+  let creator = event.params.creator
+  let to = event.params.to
+  let profileId = event.params.profileId
+
+  log.debug('event.block.number {}', [event.block.number.toString()])
```
[the remark this answers is not in the document]
Comment (sharathkrml) [reply lens/src/mappings/index.ts]: removed

File: lens/src/mappings/index.ts (hunk 2 above)
Comment (sharathkrml) [reply lens/src/mappings/index.ts]: done

**Review by rahul7668gupta — APPROVED** [review]

**Pull request merged**